flink-runtime/src/main/java/org/apache/flink/runtime/operators/GroupReduceCombineDriver.java

```diff
@@ -170,7 +171,14 @@ public void run() throws Exception {
 
 			// write the value again
 			if (!this.sorter.write(value)) {
-				throw new IOException("Cannot write record to fresh sort buffer. Record too large.");
+				if (oversizedRecordCount == Long.MAX_VALUE) {
```

Comment:
I think we can skip this test. It's going to be very hard to process 2^63 oversized records in a single Flink thread. Might take a millennium or so ;-)